staging/docker-registry/upgrade.sh

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env bash
+
+# This script upgrades the docker-registry by copying all the latest upstream helm files.
+# 
+# It then applies all the needed mesosphere changes from the /patch folder.
+#
+# To upgrade, simply run: 
+#   ./upgrade_operator.sh
+
+set -xeuo pipefail
+shopt -s dotglob
+
+BASEDIR=$(dirname "$(realpath "$0")")
+UPSTREAM_REPO=git@github.com:twuni/docker-registry.helm.git
+REF=main
```

Comment:
is `main` stable enough or should we use tags? I know with tags we would have to always update that ref here but maybe a little safer to intentionally pull in a specific version

Reply:
This repo only had a couple of releases in the last 3 years https://github.com/twuni/docker-registry.helm/releases, but is relatively active and keeps up with new image releases pretty regularly without tagging.

Unfortunately there really isn't a better maintained upstream helm chart that we can use.